src/levanter/trainer.py

```diff
@@ -104,7 +103,7 @@ class TrainerConfig:
     steps_per_eval: int = 1_000  # how often to evaluate
     max_eval_batches: Optional[int] = None  # max number of batches to evaluate on. None means all batches
 
-    checkpointer: CheckpointerConfig = CheckpointerConfig()
+    checkpointer: CheckpointerConfig = field(default_factory=CheckpointerConfig)
```

Comment:
i'll probably just go back through and make most of these frozen, which i imagine will fix.